### DISCUSSION TITLE
如何往qq发送文件？

### DISCUSSION THREAD

**Discussion by leset0ng** [nonebot/discussions#490] [问答]
我使用的是nonebot+llonebot的正向ws（其他方式貌似连不上，适配器是ai）方式使用机器人，在我的插件中我想要给用户上传文件
我问了chatgpt，它给了我一段代码，我稍微修改后，代码片段是这样的
```python
sent=on_command("sent",rule=to_me())
@sent.handle()
async def handle_function(event:Event,bot:Bot):
    file_msg = MessageSegment("file", {"file":r"./file.txt"})
    await bot.send_msg(user_id=event.get_user_id(),message=file_msg)
    await sent.finish()
``` 
但是这个会报错 `ActionFailed(status='failed', retcode=1200, data=None, message="Error: ENOENT: no such file or directory, open 'C:\\Users\\wsles\\Documents\\LiteloaderQQNT\\data\\LLOneBot\\temp\\908cbcd4-1ef9-4820-be19-44ac7019bcf6'", wording="Error: ENOENT: no such file or directory, open 'C:\\Users\\wsles\\Documents\\LiteloaderQQNT\\data\\LLOneBot\\temp\\908cbcd4-1ef9-4820-be19-44ac7019bcf6'", echo='1')`
我感觉是llonebot的问题，但是没有得到准确的回复
因此我想问一下我的方法是否正确，如果不是，能否告诉我正确的方式？

**Comment by yanyongyu** [comment]
在 ob v11 里，文件要以 file uri 的形式传递路径。即：

```python
from pathlib import Path

file_uri = Path("/path/to/your/file").as_uri()
```
